include/cudawrappers/cu.hpp

```diff
@@ -94,6 +94,22 @@ class Wrapper {
 
   explicit Wrapper(T &obj) : _obj(obj) {}
 
+  template <CUmemorytype... AllowedMemoryTypes>
+  inline void checkPointerAccess(const CUdeviceptr &pointer) const {
+    CUmemorytype memoryType;
+    checkCudaCall(cuPointerGetAttribute(
+        &memoryType, CU_POINTER_ATTRIBUTE_MEMORY_TYPE, pointer));
+
+    // Check if the memoryType is one of the allowed memory types
+    bool isAllowed = false;
+    for (auto allowedType : {AllowedMemoryTypes...}) {
+      if (memoryType == allowedType) return;
```

Comment:
Google Style rule suggests adding curly braces to improve readability and avoid errors related to ambiguous indentation or adding lines later:
```suggestion
      if (memoryType == allowedType) {
            return;
      }
```